pkg/controller/podautoscaler/horizontal.go

```diff
@@ -159,11 +159,31 @@ func (a *HorizontalController) computeReplicasForCPUUtilization(hpa *autoscaling
 	utilization := int32(*currentUtilization)
 
 	usageRatio := float64(utilization) / float64(targetUtilization)
-	if math.Abs(1.0-usageRatio) > tolerance {
-		return int32(math.Ceil(usageRatio * float64(currentReplicas))), &utilization, timestamp, nil
+	if math.Abs(1.0-usageRatio) <= tolerance {
+		// if the change was too small, ignore it
+		return currentReplicas, &utilization, utilizationInfo.OldestTimestamp, nil
 	}
 
-	return currentReplicas, &utilization, timestamp, nil
+	// for unready pods, in the case of scale up, check to see if treating those pods as having a mock
+	// CPU utilization would change things
+	if utilizationInfo.UnreadyPodsCount > 0 && usageRatio > 1.0 {
+		// check if treating unready pods as having 0 CPU usage would change things
+		newUtilization := (*currentUtilization * utilizationInfo.ReadyPodsCount) / (utilizationInfo.ReadyPodsCount + utilizationInfo.UnreadyPodsCount)
+		newUsageRatio := float64(newUtilization) / float64(targetUtilization)
+
+		// simply don't scale if the new usage ratio would mean a downscale or no scale
+		if newUsageRatio < 0 || math.Abs(1.0-newUsageRatio) <= tolerance {
+			return currentReplicas, &utilization, utilizationInfo.OldestTimestamp, nil
+		}
+
+		// set the usage ratio to scale by to our new usage ratio (which may be a bit more conservative)
+		usageRatio = newUsageRatio
+
+		// TODO: do we want the HPA to report the actual utilization, or the rectified utilization we compute here?
+		utilization = int32(newUtilization)
```

Comment:
whoops, yeah, mixed up what I was thinking about.  The pod counts will look off, but I think I agree with you here.